### ISSUE TITLE
Data races in router

### ISSUE THREAD

**Issue by Darkren** [skycoin/skywire#237]
**Describe the bug**
Found data race on `make test`:

```
==================
WARNING: DATA RACE
Read at 0x00c0003f4558 by goroutine 275:
  github.com/SkycoinProject/skywire-mainnet/pkg/router.(*RouteGroup).sendKeepAlive()
      /Users/darkrengarius/go/src/github.com/SkycoinPro/skywire-mainnet/pkg/router/route_group.go:379 +0xcd
  github.com/SkycoinProject/skywire-mainnet/pkg/router.(*RouteGroup).keepAliveLoop()
      /Users/darkrengarius/go/src/github.com/SkycoinPro/skywire-mainnet/pkg/router/route_group.go:370 +0x2e6

Previous write at 0x00c0003f4558 by goroutine 291:
  [failed to restore the stack]

Goroutine 275 (running) created at:
  github.com/SkycoinProject/skywire-mainnet/pkg/router.NewRouteGroup()
      /Users/darkrengarius/go/src/github.com/SkycoinPro/skywire-mainnet/pkg/router/route_group.go:126 +0x777
  github.com/SkycoinProject/skywire-mainnet/pkg/router.(*router).saveRouteGroupRules()
      /Users/darkrengarius/go/src/github.com/SkycoinPro/skywire-mainnet/pkg/router/router.go:375 +0x4dd
  github.com/SkycoinProject/skywire-mainnet/pkg/router.(*router).DialRoutes()
      /Users/darkrengarius/go/src/github.com/SkycoinPro/skywire-mainnet/pkg/router/router.go:234 +0xb17
  github.com/SkycoinProject/skywire-mainnet/pkg/router.Test_router_DialRoutes()
      /Users/darkrengarius/go/src/github.com/SkycoinPro/skywire-mainnet/pkg/router/router_test.go:97 +0xf58
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:909 +0x199
==================
==================
WARNING: DATA RACE
Read at 0x00c0003f4570 by goroutine 275:
  github.com/SkycoinProject/skywire-mainnet/pkg/router.(*RouteGroup).sendKeepAlive()
      /Users/darkrengarius/go/src/github.com/SkycoinPro/skywire-mainnet/pkg/router/route_group.go:379 +0x4a3
  github.com/SkycoinProject/skywire-mainnet/pkg/router.(*RouteGroup).keepAliveLoop()
      /Users/darkrengarius/go/src/github.com/SkycoinPro/skywire-mainnet/pkg/router/route_group.go:370 +0x2e6

Previous write at 0x00c0003f4570 by goroutine 291:
  [failed to restore the stack]

Goroutine 275 (running) created at:
  github.com/SkycoinProject/skywire-mainnet/pkg/router.NewRouteGroup()
      /Users/darkrengarius/go/src/github.com/SkycoinPro/skywire-mainnet/pkg/router/route_group.go:126 +0x777
  github.com/SkycoinProject/skywire-mainnet/pkg/router.(*router).saveRouteGroupRules()
      /Users/darkrengarius/go/src/github.com/SkycoinPro/skywire-mainnet/pkg/router/router.go:375 +0x4dd
  github.com/SkycoinProject/skywire-mainnet/pkg/router.(*router).DialRoutes()
      /Users/darkrengarius/go/src/github.com/SkycoinPro/skywire-mainnet/pkg/router/router.go:234 +0xb17
  github.com/SkycoinProject/skywire-mainnet/pkg/router.Test_router_DialRoutes()
      /Users/darkrengarius/go/src/github.com/SkycoinPro/skywire-mainnet/pkg/router/router_test.go:97 +0xf58
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:909 +0x199
==================
==================
WARNING: DATA RACE
Read at 0x00c0002fe0e0 by goroutine 275:
  github.com/SkycoinProject/skywire-mainnet/pkg/router.(*RouteGroup).sendKeepAlive()
      /Users/darkrengarius/go/src/github.com/SkycoinPro/skywire-mainnet/pkg/router/route_group.go:385 +0x1d3
  github.com/SkycoinProject/skywire-mainnet/pkg/router.(*RouteGroup).keepAliveLoop()
      /Users/darkrengarius/go/src/github.com/SkycoinPro/skywire-mainnet/pkg/router/route_group.go:370 +0x2e6

Previous write at 0x00c0002fe0e0 by goroutine 291:
  [failed to restore the stack]

Goroutine 275 (running) created at:
  github.com/SkycoinProject/skywire-mainnet/pkg/router.NewRouteGroup()
      /Users/darkrengarius/go/src/github.com/SkycoinPro/skywire-mainnet/pkg/router/route_group.go:126 +0x777
  github.com/SkycoinProject/skywire-mainnet/pkg/router.(*router).saveRouteGroupRules()
      /Users/darkrengarius/go/src/github.com/SkycoinPro/skywire-mainnet/pkg/router/router.go:375 +0x4dd
  github.com/SkycoinProject/skywire-mainnet/pkg/router.(*router).DialRoutes()
      /Users/darkrengarius/go/src/github.com/SkycoinPro/skywire-mainnet/pkg/router/router.go:234 +0xb17
  github.com/SkycoinProject/skywire-mainnet/pkg/router.Test_router_DialRoutes()
      /Users/darkrengarius/go/src/github.com/SkycoinPro/skywire-mainnet/pkg/router/router_test.go:97 +0xf58
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:909 +0x199
==================
==================
WARNING: DATA RACE
Read at 0x00c0004bb640 by goroutine 275:
  github.com/SkycoinProject/skywire-mainnet/pkg/router.(*RouteGroup).sendKeepAlive()
      /Users/darkrengarius/go/src/github.com/SkycoinPro/skywire-mainnet/pkg/router/route_group.go:386 +0x249
  github.com/SkycoinProject/skywire-mainnet/pkg/router.(*RouteGroup).keepAliveLoop()
      /Users/darkrengarius/go/src/github.com/SkycoinPro/skywire-mainnet/pkg/router/route_group.go:370 +0x2e6

Previous write at 0x00c0004bb640 by goroutine 291:
  [failed to restore the stack]

Goroutine 275 (running) created at:
  github.com/SkycoinProject/skywire-mainnet/pkg/router.NewRouteGroup()
      /Users/darkrengarius/go/src/github.com/SkycoinPro/skywire-mainnet/pkg/router/route_group.go:126 +0x777
  github.com/SkycoinProject/skywire-mainnet/pkg/router.(*router).saveRouteGroupRules()
      /Users/darkrengarius/go/src/github.com/SkycoinPro/skywire-mainnet/pkg/router/router.go:375 +0x4dd
  github.com/SkycoinProject/skywire-mainnet/pkg/router.(*router).DialRoutes()
      /Users/darkrengarius/go/src/github.com/SkycoinPro/skywire-mainnet/pkg/router/router.go:234 +0xb17
  github.com/SkycoinProject/skywire-mainnet/pkg/router.Test_router_DialRoutes()
      /Users/darkrengarius/go/src/github.com/SkycoinPro/skywire-mainnet/pkg/router/router_test.go:97 +0xf58
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:909 +0x199
==================
Found 4 data race(s)
FAIL	github.com/SkycoinProject/skywire-mainnet/pkg/router	74.099s
```

**Comment by Darkren** [comment]
Bump, still persists, just encountered

**Comment by jdknives** [comment]
Likely fixed by #278
